### PR TITLE
Bug fix: custom error pages should be shown on 400, 404, 500

### PR DESCRIPTION
**Pull request by skofman1**
Additional change: cleanup old unused error pages

Fixes: https://github.com/NuGet/NuGetGallery/issues/4623

### CODE REVIEW

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -294,9 +294,8 @@
       <!-- Remove Default HTTP Handler -->
       <remove path="*" verb="GET,HEAD,POST" />
     </httpHandlers>
-    <!-- Note: The Error.html file is in the ~/Public folder, but the URL Rewrite rules below take ~/Error.html and rewrite it appropriately. -->
-    <customErrors mode="RemoteOnly" defaultRedirect="~/Error.html">
-      <error statusCode="400" redirect="~/Errors/400" />
+    <customErrors mode="RemoteOnly" defaultRedirect="~/Errors/500" redirectMode="ResponseRedirect">
+      <error statusCode="400" redirect="~/Errors/400?error=1" />
       <error statusCode="404" redirect="~/Errors/404" />
```
Comment (joelverhagen) [inline src/NuGetGallery/Web.config]: Should re pl

Comment (joelverhagen) [reply src/NuGetGallery/Web.config]: These could theoretically lead of 404 and 500 as well depending on the logic of the action, right?

File: src/NuGetGallery/Web.config [hunk 2]
```diff
@@ -294,9 +294,8 @@
       <!-- Remove Default HTTP Handler -->
       <remove path="*" verb="GET,HEAD,POST" />
     </httpHandlers>
-    <!-- Note: The Error.html file is in the ~/Public folder, but the URL Rewrite rules below take ~/Error.html and rewrite it appropriately. -->
-    <customErrors mode="RemoteOnly" defaultRedirect="~/Error.html">
-      <error statusCode="400" redirect="~/Errors/400" />
+    <customErrors mode="RemoteOnly" defaultRedirect="~/Errors/500" redirectMode="ResponseRedirect">
+      <error statusCode="400" redirect="~/Errors/400?error=1" />
```
Comment (joelverhagen) [inline src/NuGetGallery/Web.config]: Spoke offline. Would `?` or `?error` work just as well?

Comment (loic-sharma) [reply src/NuGetGallery/Web.config]: Could we have a comment explaining what this hack does? :)

Comment (skofman1) [reply src/NuGetGallery/Web.config]: Apparently just ? works 🌞

File: src/NuGetGallery/Web.config (hunk 1 above)
Comment (skofman1) [reply src/NuGetGallery/Web.config]: I don't see a problem with keeping the query params for 404/500. There was a specific issue with 400 with <script></script> in the query string that causes continuous failures. This won't happen for 404/500.

**Pull request merged**